lib/src/main/java/graphql/nadel/validation/NadelHydrationValidation.kt

```diff
@@ -227,14 +239,38 @@ internal class NadelHydrationValidation(
                 if (argument == null) {
                     MissingHydrationArgumentValueSource(parent, overallField, remoteArgSource)
                 } else {
-                    // TODO: check argument type is correct
-                    null
+                    //check the input types match with hydration and actor fields
+                    val hydrationArgType = argument.type
+                    return validateHydrationInputArg(
+                            hydrationArgType,
+                            actorFieldArg.type,
+                            parent,
+                            overallField,
+                            remoteArgDef,
+                            hydration,
+                            isBatchHydration
+                    )
                 }
             }
 
-            else -> {
-                null
+            StaticArgument -> {
+                val staticArg = remoteArgSource.staticValue
+                if (!validationUtil.isValidLiteralValue(
```

Comment:
isValidLiteralValue is a public recursive function from this `graphql-java` class:
https://github.com/graphql-java/graphql-java/blob/dd50648b5e178f965918d0c352787bd81654c7fe/src/main/java/graphql/validation/ValidationUtil.java

This is an internal class, but I have checked over the code and confirmed it is what we want, including recursively checking inner types of objects/functions and everything.
Have also added extensive tests for static args to check this in NadelHydrationArgumentValidationTest.kt
It is an internal class in